### PR TITLE
Structural Changes

### PR DESCRIPTION
**Pull request by petedejoy**
This PR doesn't change any core _logic_ embedded in any modules within this repository, but it adds some key structural elements to make the package more intuitive, approachable, and reliable including:
- Adding Airflow and Great Expectations as install requirements to the package.
- Reformatting docstrings to adhere to the mechanics of [sphinx standards](https://www.sphinx-doc.org/).
- Adding an `apache_airflow_provider` entry point that allows the provider to [interact with certain Airflow APIs](https://airflow.apache.org/docs/apache-airflow-providers/index.html#how-to-create-your-own-provider) and create a more customized and well-integrated experience.
- Adding additional instructions to the Readme on using the provider and applying necessary configuration changes to make it work with Airflow 2.0.
- Refactoring the example DAG into two separate files for improved modularity.
- Fixing configuration and import errors associated with the original example DAG- it now runs in a contextual demo that shows off how the `GreatExpectationsOperator` can be extended to handle different kinds of validation checks.
- Adding a unit tests for the `GreatExpectationsOperator`
- Pulling unnecessary files out of the package, including the data context and example data sets. We've adjusted testing instructions and paths in the actual configuration files to accommodate this new file structure.

### CODE REVIEW

**Comment by petedejoy** [comment]
I've tested this pretty extensively, but if anyone wants to check me on it, here are the steps I've been following:

1. Clone my branch
2. cd into provider directory
3. Run `python3 -m pip install build`
4. Run `python3 -m build` to build the wheel
5. Find the .whl file in /dist/*.whl
6. Download the [Astro CLI](https://github.com/astronomer/astro-cli)
7. Create a new project directory, cd into it, and run `astro dev init` to initialize a new astro project
8. Ensure the Dockerfile contains Airflow 2.0: `FROM quay.io/astronomer/ap-airflow:2.0.0-buster-onbuild`
9. Copy `.whl` file to the top-level of your Astro project
10. Add to Dockerfile `RUN pip install --user airflow_provider_fivetran-0.0.1-py3-none-any.whl` to install the wheel into the containerized operating environment.
11. Copy your sample DAG to the `dags/` folder of your astro project directory.
12. Run `astro dev start` to build the containers and run Airflow locally (you'll need Docker on your machine).
13. When you're done, run `astro dev stop` to wind down the deployment (or `astro dev kill` to kill the containers and remove the local Docker volumes. You can also use this command if you need to rebuild the environment with a new .whl file.

The only error that should be observed is one with the example DAG showcasing the `GreatExpecationsBigQueryOperator`- this is due to the `get_connection` request being made from the `init` method instead of at runtime. It creates a fatal import error _if_ a connection object doesn't exist, but should work fine once that connection is added. Regardless, calls to the Airflow db shouldn't be made in `init`, so I'll open up a separate issue to resolve that one.

File: README.md [hunk 1]
```diff
@@ -13,32 +13,48 @@ An Airflow operator for [Great Expectations](greatexpectations.io), a Python lib
...
 ```
 
 ## Examples
 
-See the [**examples**](./great_expectations_provider/examples) directory for an example DAG with some sample tasks that demonstrate operator functionality. The example DAG file contains a comment with instructions on how to run the examples.
+See the [**example_dags**](https://github.com/great-expectations/airflow-provider-great-expectations/tree/main/great_expectations_provider/example_dags) directory for an example DAG with some sample tasks that demonstrate operator functionality. The example DAG file contains a comment with instructions on how to run the examples.
+
+Note that to make these operators work, you will need to change the value of [`enable_xcom_pickling`](https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg#L181) to `true` in your airflow.cfg.
+
+To run these examples:
```
Comment (spbail) [inline README.md:48]: I'm ok with giving the steps for Astro here, but it should be clear that it's not necessary. I'm thinking either have two paths:
### To run these examples:

**Option 1: Run with a local Airflow installation**
Point your dag_folder at the example_dags directory and run Airflow as usual. (or something like that)

**Option 2: Run with [Astronomer](https://www.astronomer.io)**
Steps below.

File: README.md [hunk 2]
```diff
@@ -13,32 +13,48 @@ An Airflow operator for [Great Expectations](greatexpectations.io), a Python lib
...
+1. Initialize a project with the [Astro CLI](https://www.astronomer.io/docs/cloud/stable/develop/cli-quickstart)
+2. Copy the example DAG into the `dags/` folder of your astro project
+3. Add the following env var to your `Dockerfile` to enable xcom pickling: `ENV AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True`
+4. Copy the directories in the `include` folder of this repository into the `include` directory of your Astro project
+5. Add `airflow-provider-great-expectations` to your `requirements.txt`
+6. Run `astro dev start` to view the DAG on a local Airflow instance (you will need Docker running)
+
+**This operator is in very early stages of development! Feel free to submit issues, PRs, or ping the current author (Sam Bail) in the [Great Expectations Slack](http://greatexpectations.io/slack) for feedback. Thanks to [Pete DeJoy](https://github.com/petedejoy) and the [Astronomer.io](https://www.astronomer.io/) team for the support.
```
Comment (spbail) [inline README.md:56]: >> This operator is in very early stages of development! Feel free to submit issues, PRs, or ping the current author (Sam Bail) in the [Great Expectations Slack](http://greatexpectations.io/slack) for feedback.

Rebase, I changed that :)

File: great_expectations_provider/__init__.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+def get_provider_info():
+    return {
+        "package-name": "airflow-provider-great-expectations",
+        "name": "Great Expectations Provider",
+        "description": "An Apache Airflow provider for GE.",
```
Comment (spbail) [inline great_expectations_provider/__init__.py:5]:
```suggestion
        "description": "An Apache Airflow provider for Great Expectations.",
```

Comment (spbail) [reply great_expectations_provider/__init__.py:5]: Style guide, we spell it out in docs (abbreviation is OK in inline code comments I guess)

File: great_expectations_provider/__init__.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+        "versions": ["0.0.4"]
+    }
```
Comment (spbail) [inline great_expectations_provider/__init__.py:7]:
```suggestion
    }

```

File: great_expectations_provider/example_dags/bigquery_example.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+"""
+A DAG that demonstrates implementation of the GreatExpectationsBigQueryOperator. 
+
+Note: you wil need to reference the necessary data assets and expectations suites in your project. You can find samples available in the provider source directory.
+
+Steps to run:
```
Comment (spbail) [inline great_expectations_provider/example_dags/bigquery_example.py:6]: See my comment re: defaulting to Astro. Also, for maintainability, I would prefer to say "see the steps in the README.md" and not repeat them here, and only add the bigquery connection at the end.

File: great_expectations_provider/operators/great_expectations.py [hunk 6]
```diff
@@ -47,6 +71,7 @@ def __init__(self,
                  on_failure_callback=None,
                  **kwargs
                  ):
+<<<<<<< HEAD
```
Comment (spbail) [inline great_expectations_provider/operators/great_expectations.py:74]: Leftover from merge, another one below.

File: include/great_expectations/checkpoints/taxi.fail.chk.yml [hunk 7]
```diff
@@ -14,7 +14,10 @@ validation_operator_name: action_list_operator
 # Great Expectations in your pipelines easy!
 batches:
   - batch_kwargs:
-      path: <PATH_TO>/airflow-provider-great-expectations/great_expectations_provider/examples/data/yellow_tripdata_sample_2019-02.csv
+      # Note this path is designed to integrate with an Astronomer Airflow project where the top-level
```
Comment (spbail) [inline include/great_expectations/checkpoints/taxi.fail.chk.yml:17]: The missing explanation :)

File: tests/operators/test_operators.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
+"""
```
Comment (spbail) [inline tests/operators/test_operators.py:1]: OMG tests!! \o/

We don't have CI running on this yet (see "experimental" notes in the README), but this is going to be great for development at least, thank you!

I had some comments regarding adding more tests, but I'll open an issue to add more tests.

File: tests/operators/test_operators.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+# Set relative paths for Great Expectations directory and sample data
+base_path = Path(__file__).parents[2]
+data_file = os.path.join(base_path,
+                         'include',
+                         'data/yellow_tripdata_sample_2019-01.csv')
+ge_root_dir = os.path.join(base_path, 'include', 'great_expectations')
+
+
+@mock.patch.dict('os.environ', AIRFLOW_CONN_MY_BIGQUERY_CONN_ID='http://API_KEY:API_SECRET@?extra__google_cloud_platform__key_path=extra__google_cloud_platform__key_path')
```
Comment (spbail) [inline tests/operators/test_operators.py:36]: Do we need this is we're only testing the base operator?

File: tests/operators/test_operators.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+class TestGreatExpectationsOperator(unittest.TestCase):
+    """ 
+    Test functions for GreatExpectationsOperator Operator. 
+    """
+
+    def test_great_expectations_operator(self):
```
Comment (spbail) [inline tests/operators/test_operators.py:42]:
```suggestion
    def test_great_expectations_operator_batch_kwargs_success(self):
```

The task IDs in the example basically describe the input and expected behavior, so would be nice to have the tests named that way too.

**Review by spbail — CHANGES_REQUESTED** [review]
Comments inline.

File: tests/operators/test_operators.py (hunk 9 above)
Comment (pgzmnk) [reply tests/operators/test_operators.py:36]: You're right. Let's leave it out.

File: tests/operators/test_operators.py (hunk 8 above)
Comment (pgzmnk) [reply tests/operators/test_operators.py:1]: Awesome. Please comment away! Open to suggestions. 

Currently testing with unittest like so: `python3 -m unittest test.operators.test_operators.TestGreatExpectationsOperator`

File: great_expectations_provider/operators/great_expectations.py [hunk 11]
```diff
@@ -43,7 +43,7 @@ class GreatExpectationsOperator(BaseOperator):
     :param batch_kwargs: The batch_kwargs to use for validation
     :type batch_kwargs: dict
     :param assets_to_validate: A list of dictionaries of batch_kwargs + Expectation Suites to use for validation
-    :type assets_to_validate: iterable
+    :type assets_to_validate:iterable
```
Comment (spbail) [inline great_expectations_provider/operators/great_expectations.py:46]:
```suggestion
    :type assets_to_validate: iterable
```

File: tests/operators/test_operators.py (hunk 9 above)
Comment (spbail) [inline tests/operators/test_operators.py:36]:
```suggestion
```

**Review by spbail — APPROVED** [review]
Made the changes to the tests, this looks good - I'll merge!

**Pull request merged**